catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KmzTransformerImpl.java

```diff
@@ -0,0 +1,128 @@
+ */
+public class KmzTransformerImpl implements KmzTransformer {
+
+  private static final MimeType KMZ_MIMETYPE;
+
+  private static final String DOC_KML = "doc.kml";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KmzTransformerImpl.class);
+
+  static {
+    try {
+      KMZ_MIMETYPE = new MimeType("application/vnd.google-earth.kmz");
+    } catch (MimeTypeParseException e) {
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  private KMLTransformer kmlTransformer;
+
+  public KmzTransformerImpl(KMLTransformer kmlTransformer) {
+    this.kmlTransformer = kmlTransformer;
+  }
+
+  /**
+   * Transforms a {@link SourceResponse} to a zipped KMZ file.
+   *
+   * @param upstreamResponse {@link SourceResponse} containing the results of a query.
+   * @param arguments the arguments that may be used to execute the transform
+   * @return KMZ BinaryContent
+   * @throws CatalogTransformerException
+   */
+  @Override
+  public BinaryContent transform(
+      SourceResponse upstreamResponse, Map<String, Serializable> arguments)
+      throws CatalogTransformerException {
+    BinaryContent kml = kmlTransformer.transform(upstreamResponse, arguments);
+    BinaryContent kmz = transformKmlToKmz(kml);
+
+    if (kmz == null) {
+      throw new CatalogTransformerException("Unable to transform KML to KMZ.");
+    }
+    return kmz;
+  }
+
+  /**
+   * Transforms a single metacard to a zipped KMZ file.
+   *
+   * @param metacard the {@link Metacard} to be transformed
+   * @param arguments any arguments to be used in the transformation. Keys are specific to each
+   *     {@link MetacardTransformer} implementation
+   * @return KMZ BinaryContent
+   * @throws CatalogTransformerException
+   */
+  @Override
+  public BinaryContent transform(Metacard metacard, Map<String, Serializable> arguments)
+      throws CatalogTransformerException {
+    BinaryContent kml = kmlTransformer.transform(metacard, arguments);
+    BinaryContent kmz = transformKmlToKmz(kml);
+
+    if (kmz == null) {
+      throw new CatalogTransformerException(
+          "Unable to transform to KMZ for metacard ID: " + metacard.getId());
+    }
+    return kmz;
+  }
+
+  /**
+   * Converts an unzipped KML file packaged as a {@link BinaryContent} to a zipped KMZ file.
+   *
+   * @param kml - unzipped kml {@link BinaryContent}
+   * @return BinaryContent - zipped KML file containing KML data.
+   */
+  public BinaryContent transformKmlToKmz(BinaryContent kml) {
+    try {
+      InputStream inputStream = kml.getInputStream();
+      TemporaryFileBackedOutputStream temporaryFileBackedOutputStream =
+          new TemporaryFileBackedOutputStream();
+      ZipOutputStream zipOutputStream = new ZipOutputStream(temporaryFileBackedOutputStream);
+
+      final ZipEntry e = new ZipEntry(DOC_KML);
+      zipOutputStream.putNextEntry(e);
+      IOUtils.copy(inputStream, zipOutputStream);
+      zipOutputStream.closeEntry();
+      zipOutputStream.finish();
```

Comment:
❓ Since we're not using try-with-resources and finish() doesn't close the underlying output stream - does that mean that we are leaking the resource in either the normal or error case?